include/rime.hpp

```diff
@@ -512,133 +512,61 @@ namespace rime {
 
     fn character_class(I& it, const S fin) {
       consume(it);
-      if (it == fin) {
-        // []が閉じていない
-        REGEX_PATTERN_ERROR("The range of character(character class) is not closed.");
-      }
-      
-      if (*it == chars::caret) {
+      if (it != fin and *it == chars::caret) {
         consume(it);
-        if (it == fin) {
-          // []が閉じていない
-          REGEX_PATTERN_ERROR("The range of character(character class) is not closed.");
-        }
       }
 
       class_ranges(it, fin);
-
-      if (*it == chars::rbracket) {
-        consume(it);
-        return;
-      } else {
-        // []が閉じていない
-        REGEX_PATTERN_ERROR("The range of character(character class) is not closed.");
-      }
+      consume(it);
     }
 
     // class_atomがどの構文をパースして帰っているのかを伝える
     enum class class_atom_result : unsigned char { 
```

Comment:
2値になったので`bool`でいいですね
```suggestion
    enum class class_atom_result : bool { 
```